src/test/java/com/thebois/models/inventory/ColonyInventoryTests.java

```diff
@@ -0,0 +1,124 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Stack;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+
+import com.thebois.models.IFinder;
+import com.thebois.models.beings.Colony;
+import com.thebois.models.beings.pathfinding.IPathFinder;
+import com.thebois.models.inventory.items.IItem;
+import com.thebois.models.inventory.items.ItemType;
+import com.thebois.models.inventory.items.Log;
+import com.thebois.models.inventory.items.Rock;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class ColonyInventoryTests {
+
+    private static Stream<Arguments> calculatedDifferenceEqualToExpectedSource() {
+        return Stream.of(
+            Arguments.of(List.of(
+                             new Rock(),
+                             new Rock(),
+                             new Rock(),
+                             new Log(),
+                             new Log(),
+                             new Log()),
+                         List.of(new Rock()),
+                         new ArrayList<>(Arrays.asList(ItemType.LOG,
+                                                       ItemType.LOG,
+                                                       ItemType.LOG,
+                                                       ItemType.ROCK,
+                                                       ItemType.ROCK))),
+            Arguments.of(List.of(new Rock(), new Rock()),
+                         List.of(new Rock()),
+                         new ArrayList<>(List.of(ItemType.ROCK))));
+    }
+
+    @ParameterizedTest
+    @MethodSource("calculatedDifferenceEqualToExpectedSource")
+    public void calculatedDifferenceEqualToExpected(
+        final Collection<IItem> listA,
+        final Collection<IItem> listB,
+        final ArrayList<ItemType> expectedDifference) {
+        // Arrange
+
+        final Stack<IItem> itemsA = new Stack<>();
+        final Stack<IItem> itemsB = new Stack<>();
+        itemsA.addAll(listA);
+        itemsB.addAll(listB);
+
+        final IInventory colonyA = new Colony(new ArrayList<>(),
+                                               Mockito.mock(IPathFinder.class),
+                                               Mockito.mock(IFinder.class));
+        final IInventory colonyB = new Colony(new ArrayList<>(),
+                                                          Mockito.mock(IPathFinder.class),
+                                                          Mockito.mock(IFinder.class));
+
+        colonyA.addMultiple(itemsA);
+        colonyB.addMultiple(itemsB);
+
+        // Act
+        final ArrayList<ItemType> difference = colonyA.calculateDifference(colonyB);
+
+        // Assert
+        assertThat(difference.containsAll(expectedDifference)).isTrue();
+    }
```

Comment:
question

This is an inventory method, so it should be tested in `InventoryTests` right? Or why did you place it in a specific class for just `ColonyInventoryTests`?